### PR TITLE
Add support for generating SVM relocations on Power

### PR DESCRIPTION
**Pull request by aviansie-ben**
This PR adds support to the Power codegen to generate `TR_SymbolFromManager` and `TR_DiscontiguousSymbolFromManager` relocations where necessary instead of the old relocation types when SVM is enabled.

### CODE REVIEW

**Comment by 0xdaryl** [comment]
@dsouzai, I'd appreciate a review from you please.

**Comment by dsouzai** [comment]
I'll take a look at this as well, but because this is deep in the power codegen, I'd also appreciate a review from @ymanton

**Comment by 0xdaryl** [comment]
@genie-omr build plinux

This is a Power-specific change.  A full genie build is not required.

File: compiler/p/codegen/OMRCodeGenerator.cpp [hunk 1]
```diff
@@ -3191,151 +3191,99 @@ OMR::Power::CodeGenerator::addMetaDataForLoadAddressConstantFixed(
...
+      typeAddress = TR_FixedSequenceAddress2;
 
-   bool doAOTRelocation = true;
+   TR_FixedSequenceKind seqKind;
    TR::Compilation *comp = self()->comp();
 
    if (tempReg == NULL)
+      seqKind = fixedSequence1;
+   else
+      seqKind = fixedSequence5;
```
Comment (0xdaryl) [inline compiler/p/codegen/OMRCodeGenerator.cpp]: Consider the more succinct:

`TR_FixedSequenceKind seqKind = tempReg ? fixedSequence5 : fixedSequence1;`

File: compiler/p/codegen/OMRConstantDataSnippet.cpp [hunk 2]
```diff
@@ -381,23 +382,40 @@ OMR::ConstantDataSnippet::emitAddressConstant(
...
+            {
+            TR_ExternalRelocationTargetKind kind=TR_NoRelocation;
+            if (node->isClassPointerConstant())
+               kind = TR_ClassPointer;
+            else if (node->isMethodPointerConstant())
+               {
+               if (node->getInlinedSiteIndex() == -1)
+                  kind = TR_RamMethod;
+               else
+                  kind = TR_MethodPointer;
```
Comment (0xdaryl) [inline compiler/p/codegen/OMRConstantDataSnippet.cpp]: Consider the more succinct:
```
kind = (node->getInlinedSiteIndex() == -1) ? TR_RamMethod : TR_MethodPointer;
```

File: compiler/p/codegen/OMRConstantDataSnippet.cpp [hunk 3]
```diff
@@ -381,23 +382,40 @@ OMR::ConstantDataSnippet::emitAddressConstant(
...
+               }
+
+            if (kind != TR_NoRelocation)
+               {
+               TR::Relocation *relo;
+               relo = new (cg()->trHeapMemory()) TR::ExternalRelocation(codeCursor, (uint8_t *)node, kind, cg());
```
Comment (0xdaryl) [inline compiler/p/codegen/OMRConstantDataSnippet.cpp]: Join with previous line.

**Comment by aviansie-ben** [comment]
All style issues brought up by @0xdaryl should be addressed now

**Comment by gita-omr** [comment]
@mstoodle could you please take a look?  We would like it to be merged today.

**Comment by 0xdaryl** [comment]
@genie-omr build plinux

**Pull request merged**